apps/OpenSpace/ext/launcher/src/sgctedit/windowcontrol.cpp

```diff
@@ -0,0 +1,521 @@
+        _comboMonitorSelect->setCurrentIndex(_monIndex);
+    }
+    _fullscreenButton = new QPushButton(this);
+    _fullscreenButton->setText("Set to Fullscreen");
+    _checkBoxWindowDecor = new QCheckBox("Window Decoration", this);
+    _checkBoxWindowDecor->setCheckState(Qt::CheckState::Checked);
+    _checkBoxWebGui = new QCheckBox("WebGUI only this window", this);
+    _checkBoxSpoutOutput = new QCheckBox("Spout Output", this);
+    _comboProjection = new QComboBox(this);
+    _comboProjection->addItems(_projectionTypes);
+
+    _comboQuality = new QComboBox(this);
+    _comboQuality->addItems(_qualityTypes);
+
+    {
+        _lineFovH = new QLineEdit("80.0", parent);
+        _lineFovV = new QLineEdit("50.534", parent);
+        QDoubleValidator* validatorFovH = new QDoubleValidator(-180.0, 180.0, 10);
+        _lineFovH->setValidator(validatorFovH);
+        QDoubleValidator* validatorFovV = new QDoubleValidator(-90.0, 90.0, 10);
+        _lineFovV->setValidator(validatorFovV);
+        _lineHeightOffset = new QLineEdit("0.0", parent);
+        QDoubleValidator* validatorHtOff= new QDoubleValidator(-1000000.0, 1000000.0, 12);
+        _lineHeightOffset->setValidator(validatorHtOff);
+    }
+
+    connect(
+        _size_x,
+        SIGNAL(textChanged(const QString&)),
+        this,
+        SLOT(onSizeXChanged(const QString&))
+    );
+    connect(
+        _size_y,
+        SIGNAL(textChanged(const QString&)),
+        this,
+        SLOT(onSizeYChanged(const QString&))
+    );
+    connect(
+        _offset_x,
+        SIGNAL(textChanged(const QString&)),
+        this,
+        SLOT(onOffsetXChanged(const QString&))
+    );
+    connect(
+        _offset_y,
+        SIGNAL(textChanged(const QString&)),
+        this,
+        SLOT(onOffsetYChanged(const QString&))
+    );
+    connect(
+        _comboMonitorSelect,
+        SIGNAL(currentIndexChanged(int)),
+        this,
+        SLOT(onMonitorChanged(int))
+    );
+    connect(_comboProjection,
+        SIGNAL(currentIndexChanged(int)),
+        this,
+        SLOT(onProjectionChanged(int))
+    );
+    connect(_checkBoxSpoutOutput,
+        SIGNAL(stateChanged(int)),
+        this,
+        SLOT(onSpoutSelection(int))
+    );
+    connect(
+        _checkBoxWebGui,
+        SIGNAL(stateChanged(int)),
+        this,
+        SLOT(onWebGuiSelection(int))
+    );
+    connect(_fullscreenButton, SIGNAL(released()), this, SLOT(onFullscreenClicked()));
+}
+
+QVBoxLayout* WindowControl::initializeLayout() {
+    _layoutFullWindow = new QVBoxLayout();
+    //Window size
+    QVBoxLayout* layoutWindowCtrl = new QVBoxLayout();
+
+    _labelWinNum = new QLabel();
+    _labelWinNum->setText("Window " + QString::number(_index + 1));
+    QString colorStr = "QLabel { color : ";
+    colorStr += _colorsForWindows ? _colorsForWindows[_index] : "#FFFFFF";
+    colorStr += "; }";
+    _labelWinNum->setStyleSheet(colorStr);
+
+    QHBoxLayout* layoutWinNum = new QHBoxLayout();
```

Comment:
(also for rest of this file)  see above for empty constructor braces and `this` for layouts/widgets wrt reparenting